accepted/2020/workloads/workloads.md

```diff
@@ -0,0 +1,92 @@
+
+### Composition
+
+Workloads will be composed and distributed as a set of packs. Packs are just a compressed set of files, like the [ASP.NET Core runtime]((https://www.nuget.org/packages/Microsoft.AspNetCore.App.Runtime.linux-arm64)). They are not discoverable on their own, nor do they describe the other packs that may work with or require or the various scenarios in which they can be used. Today, there are many hard-coded [versions and packages described in MSBuild targets](https://github.com/dotnet/core-sdk/blob/edbdf82041abff369ed64e5d0d6fd590e75b5328/src/redist/targets/GenerateBundledVersions.targets) in the SDK that satisfy this need. Moving forward, we will externalize this compositional information into a set of workload manifest files. Each workload will include a manifest (likely distributed as its own small pack), and use it to describe the set of versioned components that it offers. MSBuild and potentially other tools will be updated to discover and read these manifests, to the end of honoring them in the build and any other relevant operations.
+
+The target framework will be used as the primary currency in project files that establish a link to a workload. As part of a related effort, [target frameworks are being updated in .NET 5](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md) to include operating system information. This change will enable a direct link to OS-specific workloads. For example, a project that targets `net5-android` will need the .NET 5 Android workload (Xamarin.Android). A project that multi-targets `net5-android` and `net5-ios` will also need the .NET 5 iOS workload (Xamarin.iOS).
+
+The set of TFM to workload mappings will be finite, and defined by the workloads. The SDK will carry a copy of this mapping, but will defer to information carried as part of installed workloads when they present newer information.
+
+### Acquisition
+
+Workloads will be acquired in two primary ways: via a supported installation tool or the .NET CLI.
+
+Note: The term "supported installation tool" is intended to mean an installation system that exposes components and manages their dependencies. For example, Visual Studio Installer, Docker, and APT do that, while MSI and PKG don't.
+
+Install tools must follow these rules for workloads it installs:
+
+* The workload manifest is included as a required dependency for all variants of the workload (some packs might be optional).
+* All referenced packs are included, matching the version specified in the workload manifest.
+
+Users should be able to install workloads via an install tool without any sense that there is a compositional model at play.
```

Comment:
Can you clarify what this means? What is the experience that would give them a sense that there is a compositional model at play?

Reply:
The point is that they should never realize that there is a composition model at play. If they do the two things I referenced above, then they will not. Another way of saying this is that it will be deliver workloads via a supporter installation tool the wrong way, and then everything will break, at which point it will be obvious that there is a compositional model at play, because the product will produce poorly defined and unintelligible errors.